core/src/main/clojure/xtdb/logical_plan.clj

```diff
@@ -127,10 +127,7 @@
         {:keys [col-types ->cursor stats]} (f left-col-types right-col-types)]
```

Comment:
Should the `unery-expr` above also pass to the the style of the `binary-expr`?

Reply:
tbh, given how little `binary-expr` now does for join, I'd maybe have opted for join bypassing it entirely, so that the set operators can continue to use the current version.

context is that previously (before we had `with-close-on-catch`) this did way more - imagine inlining `with-close-on-catch` in all of the call-sites :sweat_smile: